src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -13,55 +13,43 @@
 import static net.openhft.hashing.Util.*;
 
 /**
- * Tuple hash function producing {@code long[]} result from byte sequences of any length and
- * a plenty of different sources which "feels like byte sequences". Except {@link
- * #hashBytes(byte[])}, {@link #hashBytes(ByteBuffer)} (with their "sliced" versions) and
- * {@link #hashMemory(long, long)} methods, which actually accept byte sequences, notion of byte
- * sequence is defined as follows:
+ * Tuple hash function producing more than 64-bit hash code into a result array of type
+ * {@code long[]} from any byte sequences. See {@link LongHashFunction} for the definition of byte
+ * sequence semantics and {@link ByteOrder} requirements for implementations.
+ *
+ * Every {#link LongHashFunction} hash method has two corresponding ones in this class for different
+ * allocation stradges:
  * <ul>
- *     <li>For methods accepting arrays of Java primitives, {@code String}s and
- *     {@code StringBuilder}s, byte sequence is how the input's bytes are actually lay in memory.
+ *     <li>{@code void hash(input..., long[] result)} will store the hash results in array
+ *     {@code result[0 .. newResultArray().length-1]}, and throws exceptions when {@code result}
+ *     is not valid. {@link #newResultArray} method should always be used to create resuable result
+ *     arrays to avoid exceptions. See {@link #hash(long, long[])}.</li>
+ *     <li>{@code long[] hash(input...)} will allocate and return an array containing the results.
  *     </li>
- *     <li>For methods accepting single primitive values, byte sequence is how this primitive
- *     would be put into memory with {@link ByteOrder#nativeOrder() native} byte order, or
- *     equivalently, {@code hashXxx(primitive)} has always the same result as {@code
- *     hashXxxs(new xxx[] {primitive})}, where "xxx" is any Java primitive type name.</li>
- *     <li>For {@link #hash(Object, Access, long, long)} method byte sequence abstraction
- *     is defined by the given {@link Access} strategy to the given object.</li>
  * </ul>
  *
- * <p>Tuple hash function implementation could either produce equal results for equal input on platforms
- * with different {@link ByteOrder}, favoring one byte order in terms of performance, or different
- * results, but performing equally good. This choice should be explicitly documented for all
- * {@code LongTupleHashFunction} implementations.
+ * <b>Warning:</b> Allocations <strong>cannot</strong> be totally avoid to exchange a more than
```

Comment:
I'd remove this part, maybe somebody uses the library in Valhalla build :) In general, Javadoc is not well suited for such discussions